util/filesort/filesort_test.go

```diff
@@ -95,15 +84,15 @@ func (s *testFileSortSuite) TestLessThan(c *C) {
 		{[]types.Datum{d1, d1}, []types.Datum{d0, d0}, []bool{false, false}, false},
 	}
 
-	for _, t := range tblTwoColumns {
-		ret, err := lessThan(sc, t.Arg1, t.Arg2, t.Arg3)
-		c.Assert(err, IsNil)
-		c.Assert(ret, Equals, t.Ret)
+	for _, T := range tblTwoColumns {
```

Comment:
You may try to name the struct as `testCases`, iteratee as `testCase`, for expressive naming.

```go
testCases := []struct {
		datum0 []types.Datum
		datum1 []types.Datum
		byDesc []bool
		res  bool
	}
```

Reply:
ditto above, you can join these two cases.

Reply:
Should I try to differentiate the two different struct testCases by naming the first and second as  `testCasesTblOneColumn` and `testCasesTblTwoColumns` respectively?

Reply:
You may think of it. My opinion is that they can be the same testCases array.

Preciously, I mean something like

```go
func TestLessThan(t *testing.T) {
	t.Parallel()

	sc := new(stmtctx.StatementContext)

	d0 := types.NewDatum(0)
	d1 := types.NewDatum(1)

	tests := []struct {
		i []types.Datum
		j []types.Datum
		byDesc []bool
		res  bool
	}{
		// one column
		{[]types.Datum{d0}, []types.Datum{d0}, []bool{false}, false},
		{[]types.Datum{d0}, []types.Datum{d1}, []bool{false}, true},
		{[]types.Datum{d1}, []types.Datum{d0}, []bool{false}, false},
		{[]types.Datum{d0}, []types.Datum{d0}, []bool{true}, false},
		{[]types.Datum{d0}, []types.Datum{d1}, []bool{true}, false},
		{[]types.Datum{d1}, []types.Datum{d0}, []bool{true}, true},

		// two columns
		{[]types.Datum{d0, d0}, []types.Datum{d1, d1}, []bool{false, false}, true},
		{[]types.Datum{d0, d1}, []types.Datum{d1, d1}, []bool{false, false}, true},
		{[]types.Datum{d0, d0}, []types.Datum{d1, d1}, []bool{false, false}, true},
		{[]types.Datum{d0, d0}, []types.Datum{d0, d1}, []bool{false, false}, true},
		{[]types.Datum{d0, d1}, []types.Datum{d0, d1}, []bool{false, false}, false},
		{[]types.Datum{d0, d1}, []types.Datum{d0, d0}, []bool{false, false}, false},
		{[]types.Datum{d1, d0}, []types.Datum{d0, d1}, []bool{false, false}, false},
		{[]types.Datum{d1, d1}, []types.Datum{d0, d1}, []bool{false, false}, false},
		{[]types.Datum{d1, d1}, []types.Datum{d0, d0}, []bool{false, false}, false},
	}

	for _, test := range tests {
		res, err := lessThan(sc, test.i, test.j, test.byDesc)
		require.NoError(t, err)
		require.Equal(t, test.res, res)
	}
}
```